### PR TITLE
Update merge criteria queries

### PR DESCRIPTION
**Pull request by jmguzik**
Output:
```
Weird staff-eng-approved query: [openshift/cluster-api]
Weird complement query (without 4.11): [openshift/dpu-network-operator]
Weird complement query (without 4.11): [openshift/machine-api-provider-aws]
Weird complement query (without 4.11): [openshift/machine-api-provider-azure]
Weird complement query (without 4.11): [openshift/machine-api-provider-gcp]
Weird complement query (without 4.11): [openshift/machine-os-images]
Weird complement query (without 4.11): [openshift/cluster-api-provider-kubevirt]
```
but they seem to be ok.

Signed-off-by: Jakub Guzik <jguzik@redhat.com>

### CODE REVIEW

**Comment by petr-muller** [comment]
> Weird staff-eng-approved query: [openshift/cluster-api]
> Weird complement query (without 4.11): [openshift/dpu-network-operator]
> Weird complement query (without 4.11): [openshift/machine-api-provider-aws]
> Weird complement query (without 4.11): [openshift/machine-api-provider-azure]
> Weird complement query (without 4.11): [openshift/machine-api-provider-gcp]
> Weird complement query (without 4.11): [openshift/machine-os-images]
> Weird complement query (without 4.11): [openshift/cluster-api-provider-kubevirt]

They *do* look weird though - let me try to comment inline

**Comment by petr-muller** [comment]
openshift/cluster-api-provider-kubevirt does not use OCP queries at all,  @davidvossel @nunnatsa is this correct? Should you *not* follow OCP development process (requiring approvals for backports, respecting code/feature freeze, etc)

https://github.com/openshift/release/blob/ff4825d16845cf04a748a8727b2b6de623bd4b8f/core-services/prow/02_config/openshift/cluster-api-provider-kubevirt/_prowconfig.yaml#L18-L31

File: core-services/prow/02_config/openshift/machine-os-images/_prowconfig.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 tide:
   queries:
   - includedBranches:
-    - openshift-4.10
-    - release-4.10
+    - openshift-4.11
+    - release-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/machine-os-images/_prowconfig.yaml:5]: Looks like we'll need to introduce the GA'd release query in this file (one that requires cherry-pick-approved, backport-risk-assessed) and also add 4.11 branches to the complement query

File: core-services/prow/02_config/openshift/machine-api-provider-gcp/_prowconfig.yaml [hunk 2]
```diff
@@ -18,8 +18,8 @@ tide:
     repos:
     - openshift/machine-api-provider-gcp
   - includedBranches:
-    - openshift-4.10
-    - release-4.10
+    - openshift-4.11
+    - release-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/machine-api-provider-gcp/_prowconfig.yaml:22]: We need to add 4.11 to the complement query in this config

File: core-services/prow/02_config/openshift/machine-api-provider-azure/_prowconfig.yaml [hunk 3]
```diff
@@ -18,8 +18,8 @@ tide:
     repos:
     - openshift/machine-api-provider-azure
   - includedBranches:
-    - openshift-4.10
-    - release-4.10
+    - openshift-4.11
+    - release-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/machine-api-provider-azure/_prowconfig.yaml:22]: We need to add 4.11 to the complement query in this config

File: core-services/prow/02_config/openshift/machine-api-provider-aws/_prowconfig.yaml [hunk 4]
```diff
@@ -18,8 +18,8 @@ tide:
     repos:
     - openshift/machine-api-provider-aws
   - includedBranches:
-    - openshift-4.10
-    - release-4.10
+    - openshift-4.11
+    - release-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/machine-api-provider-aws/_prowconfig.yaml:22]: We need to add 4.11 to the complement query in this config

File: core-services/prow/02_config/openshift/dpu-network-operator/_prowconfig.yaml [hunk 5]
```diff
@@ -18,8 +18,8 @@ tide:
     repos:
     - openshift/dpu-network-operator
   - includedBranches:
-    - openshift-4.10
-    - release-4.10
+    - openshift-4.11
+    - release-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/dpu-network-operator/_prowconfig.yaml:22]: We need to add 4.11 to the complement query in this config

File: core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml [hunk 6]
```diff
@@ -72,6 +72,8 @@ tide:
     - main
     - master
     - openshift-4.1
+    - openshift-4.10
+    - openshift-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml:76]: Ok this config file was for some reason wrong. This update to the complement query is fine, but the `staff-eng-approved` one needs to have `4.10`  branches and the `cherry-pick-aproved` one needs to have 4.9 and 4.10 ones added.

File: core-services/prow/02_config/openshift/machine-api-provider-gcp/_prowconfig.yaml [hunk 7]
```diff
@@ -54,6 +54,8 @@ tide:
     - main
     - master
     - release-4.10
+    - release-4.11
+    - openshift-4.11
```
Comment (jmguzik) [inline core-services/prow/02_config/openshift/machine-api-provider-gcp/_prowconfig.yaml:58]: @petr-muller is that what you meant by adding to complement query?

Comment (petr-muller) [reply core-services/prow/02_config/openshift/machine-api-provider-gcp/_prowconfig.yaml:58]: Yes. Check out https://docs.google.com/document/d/19kxmzXFnXrbLChZXBfxy68mCYg06j3qT93VF23wepog/edit#heading=h.nzgnzwm8ihp5 - the [complement](https://en.wikipedia.org/wiki/Complement_(set_theory)) query should basically contain "everything else", so its `excludedBranches` list needs to contain all branhces that are included in all other queries.

File: core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml [hunk 8]
```diff
@@ -34,7 +38,9 @@ tide:
     repos:
     - openshift/cluster-api
   - includedBranches:
+    - openshift-4.10
     - openshift-4.9
+    - release-4.10
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml:43]: This does not seem right. This is the `staff-eng-query`, it should only contain 4.11 branches.

File: core-services/prow/02_config/openshift/machine-os-images/_prowconfig.yaml (hunk 1 above)
Comment (petr-muller) [reply core-services/prow/02_config/openshift/machine-os-images/_prowconfig.yaml:5]: This still needs to be addressed

File: core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml (hunk 8 above)
Comment (jmguzik) [reply core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml:43]: your comment from above: 

> This update to the complement query is fine, but the staff-eng-approved one needs to have 4.10 branches

Comment (petr-muller) [reply core-services/prow/02_config/openshift/cluster-api/_prowconfig.yaml]: Yeah, sorry for that. Either a typo or I got confused.

File: core-services/prow/02_config/openshift/machine-os-images/_prowconfig.yaml [hunk 9]
```diff
@@ -35,7 +35,9 @@ tide:
   - excludedBranches:
     - main
     - openshift-4.10
+    - openshift-4.11
     - release-4.10
+    - release-4.11
```
Comment (petr-muller) [inline core-services/prow/02_config/openshift/machine-os-images/_prowconfig.yaml:40]: This is just one (second) part of my initial comment:

> Looks like we'll need to introduce the GA'd release query in this file (one that requires cherry-pick-approved, backport-risk-assessed) and also add 4.11 branches to the complement query

We still need to add that GA'd release query with 4.10 branches

**Review by petr-muller — APPROVED** [review]
Looks good now.

**Pull request merged**